### PR TITLE
Add registry config creation command

### LINKED ISSUE #858: Add registry-login command from ofc-bootstrap
In faasd, when you need to get a private registry configuration, it requires a Docker registry secret that's unencrypted, but Docker Desktop puts passwords in the keychain.

## Expected Behaviour

```
faas-cli registry-login \
  --username <your-registry-username> \
  --password-stdin
# (the enter your password and hit return)
```

## Current Behaviour

Available in ofc-bootstrap:

```
ofc-bootstrap registry-login \
  --username <your-registry-username> \
  --password-stdin
# (the enter your password and hit return)
```

## Possible Solution

Given that this is now useful for faasd as well as OFC - we can promote it into the main CLI.

I'm open to suggestions on whether `registry-login` is the correct name / wording.

## Steps to Reproduce (for bugs)

1. Get a private registry
2. Try `docker login` on Mac or Windows and see that your ~/.docker/config.json doesn't have a secret in it, but a reference to a keychain
3. Copy over to the faasd instance, and see that it fails

## Context

It seems odd to install ofc-bootstrap for this task, and OpenFaaS Cloud is likely going to be deprecated, if not put on hold for the time being.

### PR DESCRIPTION
**Pull request by nitishkumar71**
Signed-off-by: Nitishkumar Singh <nitishkumarsingh71@gmail.com>

## Description

We are including support of generating correct registry authentication file by accepting username and password from the command line.
## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Closes #858 
## How Has This Been Tested?

1. Tried to push image to private repo with existing .docker/config.json which resulted in 
```
errors:
denied: requested access to the resource is denied
unauthorized: authentication required
```
2. Generated config.json for private repository
3. replaced .docker/config.json with earlier generated config.json in step2
4. pushed the image successfully to private repo

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
@nitishkumar71 thanks for the PR.

@Waterdrips this is lifted from the ofc-bootstrap CLI, please can you review?

File: commands/registry_login.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+	registryLoginCommand.Flags().Bool("ecr", false, "If we are using ECR we need a different set of flags, so if this is set, we need to set --username and --password")
+	registryLoginCommand.Flags().String("account-id", "", "Your AWS Account id")
+	registryLoginCommand.Flags().String("region", "", "Your AWS region")
+
+	faasCmd.AddCommand(registryLoginCommand)
+}
+
+func generateRegistryAuthFile(command *cobra.Command, _ []string) error {
+	ecrEnabled, _ := command.Flags().GetBool("ecr")
```
Comment (Waterdrips) [inline commands/registry_login.go:40]: We need error handling on these flags

File: commands/registry_login.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+			fmt.Sprintf("%s.dkr.ecr.%s.amazonaws.com", accountID, region): "ecr-login",
+		},
+	}
+
+	registryBytes, err := json.MarshalIndent(data, "", " ")
+
+	return registryBytes, err
+}
+
+func writeFileToOFCTmp(fileBytes []byte) error {
```
Comment (Waterdrips) [inline commands/registry_login.go:129]: OFCTmp is only relevant to `ofc-bootstrap` and this could have a better name

**Review by Waterdrips — CHANGES_REQUESTED** [review]

File: commands/registry_login.go (hunk 1 above)
Comment (nitishkumar71) [reply commands/registry_login.go:80]: Thanks for mentioning, have handled error check in `PreRunE`

File: commands/registry_login.go (hunk 2 above)
Comment (nitishkumar71) [reply commands/registry_login.go]: Have renamed the function to match faas cli

**Comment by alexellis** [comment]
@Waterdrips when you have time, there's a new commit for review.

**Review by Waterdrips — APPROVED** [review]

**Review by alexellis — APPROVED** [review]
Let's ship it and iterate if we need to. Thank you both.

**Comment by alexellis** [comment]
Hi guys, there was a bug in this command which meant it failed the first time I tried it.

Here's the fix, please can you both take a look?

https://github.com/openfaas/faas-cli/commit/6b5e7a14a598527063c7c05fb4187739b6eb6d38

A simple end to end test of the CLI should have shown this, as I saw it typing in `faas-cli registry-login --password-stdin`

Alex

**Pull request merged**